expression/util.go

```diff
@@ -177,6 +177,31 @@ func ExtractEquivalenceColumns(result [][]Expression, exprs []Expression) [][]Ex
 	return result
 }
 
+// FindUpperBound looks for column < constant or column <= constant and returns both the column
```

Comment:
Should we support the patten like `constant > column` or `constant >= column`.

Reply:
Theoretically these are applicable but did not cover them since the common pattern is < and <= but we can easily add this and other cases like "= 1"